source/features/fault.cc

```diff
@@ -477,8 +479,25 @@ namespace WorldBuilder
       // todo: explain
       const double starting_radius = natural_coordinate.get_depth_coordinate() + depth - starting_depth;
 
+      double buffer_zone = 0.;
+
+      // Here, we check if all the coordinate points describing a fault lie within a bounding box. To do this, we use min and
+      // max points for a fault and add a buffer zone surrounding it. Since the fault extends over a depth (i.e., fault lenght),
+      // the idea is to use a small buffer zone using the approximate length of fault, such that the box covers the fault
+      // extent in depth. For spherical system, we assume that the equivalent buffer is spherical distance computed at the equator.
+
+      if (world->parameters.coordinate_system->natural_coordinate_system() == CoordinateSystem::spherical)
+        buffer_zone = 2 * (maximum_fault_thickness + maximum_total_fault_length) / (const_pi * starting_radius* 2);
+      else if (world->parameters.coordinate_system->natural_coordinate_system() == CoordinateSystem::cartesian)
+        buffer_zone = 2 * (maximum_fault_thickness + maximum_total_fault_length);
```

Comment:
Can you explain why you add `maximum_total_fault_length` to the buffer zone, and why you multiply by two? The buffer zone should just be a distance around points so that the bounding box is around these points, but that buffer zone should be independent of where the points actually lie. (I'm assuming that `maximum_total_fault_length` is something that incorporated information about the points of the fault, right?)

Reply:
So, the way I implemented the bounding_box is a 2D box that contains all the possible surface points for which the composition is calculated. (i.e., the last if condition in ln 499 in ```faults.cc: WorldBuilder::Utilities::bounding_box_contains_point (...)``` The condition for the depth is satisfied in ln 498 in the same if statement.
To calculate the bounding box, I use two inputs: ``` coordinates``` and a ```buffer_zone```. The```coordinates``` is the vector of 2D points that define the fault line. Worldbuilder creates the fault using the ```coordinates, fault thickness, dip angle, and fault_length```. Using the min and max points in ```coordinates```,  we create the two diagonal points of the box: lower left and upper right. But, since the thickness of fault is not incorporated in the ```coordinates```, we need to add that to the two extreme points. Additionally, if the faults are dipping, we need to search laterally so that the surface projection of the fault is also covered in the bounding_box. For any value of dip > 0, the surface projection would not extend beyond the ```maximum_total_fault_length```.  Therefore, I add the thickness and the fault to the existing surface coordinates. The factor of 2 is somewhat arbitrary to calculate the bounding lat/long coordinates in a spherical system (I need to remove it for cartesian). If the faults lie on a great circle, we don't need any factor, and as they go towards the pole, the factor should increase.  But, I need to discuss more with you and @MFraters  on how to correctly use this.

Reply:
@alarshi Thanks for the explanation -- although I have to admit that I understand only half of it because I'm not familiar with the way the WorldBuilder describes fault, and so I have no idea what many of the words mean :-)

I very frequently look for bugs by looking for things that appear in one expression but don't go together. In this particular case, it is "thickness" and "length". I got confused why one would want to add these together because I think of one as an "extent across the fault" and the other as an "extend along the fault", and these seem conceptually independent. I can think of a very short fault (very small length) that requires a large buffer zone because it is deep, thick, and with a small dip angle. I can also think of a very long fault that requires a small buffer zone because it is shallow, not very thick, and dips vertically. So it seems wrong that the length of the fault appears in the buffer zone width. But maybe I'm also misunderstanding what WB considers the "length" of a fault :-)

Reply:
@bangerth: It is confusing because I was trying to make the bounding_box work for my case without thinking through all the details. You are right, the ```fault_length``` and ```thickness``` are perpendicular. My thinking was that if we have a vertical fault, it is sufficient that the buffer zone contains just the fault_thickness because the fault_length will be checked by the depth condition. On the other hand, a fault almost horizontal (dipping along the X-axis) would have surface coordinates depending on both thickness and the fault length, therefore, requiring a conservative estimate of buffer_zone containing both the values. 
I might be thinking it wrong, it would help if I could discuss this with you sometime today. I also want to discuss how to implement the BoundingBox class defined PR#306 for our case. :)